File: java/src/test/java/it/giannibombelli/wsc2026/testsupport/events/CapturingSubscriber.java
package it.giannibombelli.wsc2026.testsupport.events;

import it.giannibombelli.wsc2026.common.application.events.EventSubscriber;

import java.util.ArrayList;
import java.util.List;

public final class CapturingSubscriber<T> implements EventSubscriber<T> {
    private final List<T> events = new ArrayList<>();

    @Override
    public void on(T event) {
        events.add(event);
    }

    public List<T> events() {
        return List.copyOf(events);
    }
}

File: typescript/tests/testsupport/events/capturingSubscriber.ts
// Subscriber di test che cattura gli eventi ricevuti.

import type { EventSubscriber } from '@/common/application/events/eventSubscriber.js';

export class CapturingSubscriber<T> implements EventSubscriber<T> {
  private readonly _events: T[] = [];

  on(event: T): void {
    this._events.push(event);
  }

  events(): T[] {
    return [...this._events];
  }
}
